server/list.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"fmt"
+)
+
+const (
+	// MyListKey is the key used to store the order of the owned todos
+	MyListKey = ""
+	// InListKey is the key used to store the order of received todos
+	InListKey = "_in"
+	// OutListKey is the key used to store the order of sent todos
+	OutListKey = "_out"
+)
+
+// ListStore represents the KVStore operations for lists
+type ListStore interface {
+	AddItem(item *Item) error
+	GetItem(itemID string) (*Item, error)
+	RemoveItem(itemID string) error
+
+	GetItemOrder(userID string, itemID string, listID string) (*OrderElement, int, error)
+	GetItemListAndOrder(userID string, itemID string) (string, *OrderElement, int)
+
+	Add(userID string, itemID string, listID string, foreignUserID string, foreignItemID string) error
+	Remove(userID string, itemID string, listID string) error
+	Pop(userID string, listID string) (*OrderElement, error)
+
+	GetList(userID string, listID string) ([]*OrderElement, error)
+}
+
+type listManager struct {
+	store       ListStore
+	getUserName func(string) string
```

Comment:
I assume you need the `plugin.API`, that's why you're passing in the function, right? You can just store a reference to the `plugin.API` here instead, and then ListManager can either have it's own call to get the userName, or share a utility function. Usually in Go we don't store functions for that reason. (We would store them for other reasons, like a swappable algorithm, aka, the "Strategy" pattern.)

Reply:
I thought passing the API here was a bit of overkill, because we just need a function. Also, I did not want to repeat code redefining "getUserName". Nevertheless, you are right about sharing the utility function. I will do that.

Reply:
It's totally fine for the ListManager to hold a ref to the API -- we tend to pass that around whenever we need it.